include/mvdtool/bits/tsv_misc.hpp

```diff
@@ -0,0 +1,171 @@
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#pragma once
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <unordered_map>
+
+#include <boost/range/combine.hpp>
+
+#include "../tsv_except.hpp"
+
+namespace TSV {
+
+inline std::vector<TSVInfo> TSVFile::getTSVInfos() {
+    std::vector<TSVInfo> tsvFileInfoVec;
+    for (auto it_infos: tsvFileInfo) {
+        tsvFileInfoVec.push_back(it_infos.second);
+    }
+    return tsvFileInfoVec;
+}
+
+inline std::vector<TSVInfo> TSVFile::getTSVInfos(const std::vector<std::string>& me_combos, const std::vector<std::string>& morphologies) const {
+    std::vector<TSV::TSVInfo> tsvInfos;
+    for (const auto& mecombo_tupple: boost::combine(me_combos, morphologies)) {
+        auto hasMEType = tsvFileInfo.find(std::make_pair(boost::get<0>(mecombo_tupple), boost::get<1>(mecombo_tupple)));
+        if (hasMEType != tsvFileInfo.end()) {
+            tsvInfos.push_back(hasMEType->second);
+        } else {
+            std::ostringstream ss;
+            ss << "me_combo " << boost::get<0>(mecombo_tupple) << " or morphology " << boost::get<1>(mecombo_tupple) << " not found in "
+               << _filename << std::endl;
+            throw TSVException(ss.str());
+        }
+    }
+    return tsvInfos;
+}
+
+inline std::vector<boost::int32_t> TSVFile::getLayers(const std::vector<std::string>& me_combos, const std::vector<std::string>& morphologies) const {
+    std::vector<boost::int32_t> tsvLayers;
+    std::vector<TSV::TSVInfo> tsvInfos = getTSVInfos(me_combos, morphologies);
+    for (const auto& tsvInfo: tsvInfos) {
+        tsvLayers.push_back(tsvInfo.layer);
+    }
+    return tsvLayers;
+}
+
+inline std::vector<std::string> TSVFile::getMtypes(const std::vector<std::string>& me_combos, const std::vector<std::string>& morphologies) const {
+    std::vector<std::string> tsvMtypes;
+    std::vector<TSV::TSVInfo> tsvInfos = TSVFile::getTSVInfos(me_combos, morphologies);
+    for (const auto& tsvInfo: tsvInfos) {
+        tsvMtypes.push_back(tsvInfo.fullMType);
+    }
+    return tsvMtypes;
+}
+
+inline std::vector<std::string> TSVFile::getEtypes(const std::vector<std::string>& me_combos, const std::vector<std::string>& morphologies) const {
```

Comment:
Same here (generic func if possible)
Maybe we don't even need the specific accessors if this is not public API